### ISSUE TITLE
False Negative in EmptyStatement Check

### ISSUE THREAD

**Issue by Atharv3221** [checkstyle/checkstyle#18590]
I have read check documentation: https://checkstyle.sourceforge.io/checks/coding/emptystatement.html#EmptyStatement
I have downloaded the latest checkstyle from: https://github.com/checkstyle/checkstyle/releases/tag/checkstyle-13.0.0
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="EmptyStatement"/>
  </module>
</module>

```

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Test.java
public class Test {
    ; // expected violation
    ; // expected violation
    ; // expected violation

    public void name() {
        ; // violation
        ; // violation

        for (int i = 0; i < 6; i++) {
            ; // violation
            ; // violation
        }
    }
}

```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-13.0.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/atharv/test/Test.java:7:9: Empty statement. [EmptyStatement]
[ERROR] /home/atharv/test/Test.java:8:9: Empty statement. [EmptyStatement]
[ERROR] /home/atharv/test/Test.java:11:13: Empty statement. [EmptyStatement]
[ERROR] /home/atharv/test/Test.java:12:13: Empty statement. [EmptyStatement]
Audit done.
Checkstyle ends with 4 errors.

```

**Comment by Atharv3221** [comment]
@romani are they part of Declaration so not counted as statements ?

**Comment by romani** [comment]
lets cover such cases by this Check.

Please add such cases to Example to let user see behavior and make it clear we did this explicitly

**Comment by Atharv3221** [comment]
Okay

**Comment by Anushreebasics** [comment]
I am on it

**Comment by Anushreebasics** [comment]
@Atharv3221 could you please review this PR #18905
